src/main/java/com/facebook/bitEncoderDecoder/bitwise/BitwiseEncoderImpl.java

```diff
@@ -32,4 +32,23 @@ private String getBinaryString(String input) {
             return binary.toString();
         }
     }
+
+    private String checkIfEven(String triplet) {
+        StringBuilder result = new StringBuilder(triplet);
+        int sum = 0;
+        for (int i = 0; i < triplet.length(); i++) {
+            sum += Integer.parseInt(String.valueOf(triplet.charAt(i)));
+        }
+        if (sum % 2 == 0) {
+            result.append("0");
+
+        } else {
+            result.append("1");
+        }
+        while (result.length() < 4) {
+            result.append('0');
```

Comment:
appending zeros occur for bits 1-3, before counting parity bits. Parity bits are always on last positions (7-8).